Add-ons/UmbracoForms/Developer/Healthchecks/index.md

```diff
@@ -0,0 +1,117 @@
+
+In particular, prior to version 8.7, there were a number of tables that weren't defined as strictly as they should be in this area.  So we've added some primary key, foreign key and unique constraints with this version.  If you've been running a version prior to this and are upgrading, these schema updates will be applied automatically _unless_ there is existing data in the tables that prevent them from being added.
+
+There shouldn't be - but without these constraints in place it's always possible for an application bug to exist that allows for example the creation of duplicate records, or the ophaning of records, that aren't correct.  Of course, this is the reason for the constraints to exist, and why we want to ensure they are in place.
+
+### Running The Health Check
+
+To run the health check, navigate to _Settings > Health Check_ and click on the _Forms_ button.  Then click _Check Group_.
+
+You'll see a result that looks something like this:
+
+![Umbraco Forms Health Check](images/healthcheck.png)
+
+If you have a full set of green ticks, then you're all good - and no need to read on!
+
+If you have one or more red crosses though, that means a particular constraint wasn't able to be applied via the automatic schema migrations when you installed a new version of Umbraco Forms, due to existing data issues.
+
+It isn't essential that they are resolved - the package can and does function correctly without them - but for reasons of ensuring data integrity and performance, it is recommended that they are.
+
+### Resolving Reported Problems
+
+When Umbraco Forms installs an upgrade, it will attempt to apply any schema changes. If though, the update isn't essential, and it can't proceed due to existing data integrity issues, the failed update will be logged and then the rest of the migration will continue.
+
+As well as in the log files, such issues will be visible via the health check and will need to be resolved by applying scripts directly to the database.
+
+To support this, we provide the following SQL scripts:
+
+- Apply database integrity schema changes for 8.7.0 - [8.7.0-apply-keys-and-indexes.sql](scripts/8.7.0-apply-keys-and-indexes.sql)
+- Apply database integrity schema changes for 8.7.0 (forms in database tables) - [8.7.0-apply-keys-and-indexes-forms-in-db.sql](scripts/8.7.0-apply-keys-and-indexes-forms-in-db.sql)
+
+The first of these provides the SQL statements required to apply the schema updates for 8.7.0 to the common Umbraco Forms tables. The second applies just to those tables used for when forms are stored in the database, and hence only need to be applied if that option is configured.
+
+Before running any scripts or queries, please be sure to have a database backup in place.
```

Comment:
Added this to a note block in an attempt to highlight the fact that it's important to remember to have a backup in place.

```suggestion
:::note
Before running any scripts or queries, please be sure to have a database backup in place.
:::
```